src/lib/core/WeaveServerBase.cpp

```diff
@@ -191,6 +191,55 @@ WEAVE_ERROR WeaveServerBase::SendStatusReport(ExchangeContext *ec, uint32_t stat
     return err;
 }
 
+/**
+ * Indicates if the session with the given node id and the session key id is
+ * authorized to retrieve secret credential information.
+ *
+ * @return Returns 'true' if the given peer is privileged, else
+ * 'false'.
+ */
+
+bool WeaveServerBase::IsSessionPrivileged(uint16_t keyId, uint64_t peerNodeId) const
```

Comment:
I think this notion of a "privileged" session is too generic, and should not be incorporated into WeaveServerBase.  Since the current goal is to enable some very specific privileges in the context of network provisioning and fabric provisioning, I think the logic behind tracking and acting on these privileges should be in the respective server classes, and named specifically to their purpose.

For example, for network provisioning, I suggest we use names such as HasCredentialAccessPrivilege(), GrantCredentialAccessProvilege(), ClearCredentialAccessPrivilege() and mCredentialAccessSession.  A similar pattern would apply in the FabricProvisioningServer; e.g. HasFabricConfigAccessPrivilege(), etc.

Reply:
I had a bit of doubt when putting it in WeaveServerBase for the same reason but kept them there to, essentially, re-use the code for the derived provisioning servers. Will move them to the respective server classes and update.